### ISSUE TITLE
Skip query throws a 500 error, how to get additional detail?

### ISSUE THREAD

**Issue by pmkane** [openark/orchestrator#1357]
Hi there:

We are running Orchestrator 3.2.4 using pre-compiled binaries in a Percona MySQL 5.7 master-master (single writer) cluster, with a number of read-only slaves hanging off the master.  We are using GTID for replication.

Orchestrator accurately represents the infrastructure.

We are trying to skip a query on one of the read-only slaves, but it fails with a 500 error.  Unfortunately, --debug/--stack/--verbose do not provide any additional details.  The log entries we get after running skip query on the slave are:

[martini] Started GET /api/skip-query/ip-10-50-2-33/3306 for 10.50.2.62, 10.50.1.230
2021-05-19 16:04:04 DEBUG Skipping one query on ip-10-50-2-33:3306
2021-05-19 16:04:04 INFO auditType:skip-query instance:ip-10-50-2-33:3306 cluster:ip-10-50-2-67:3306 message:Skipped one query
2021-05-19 16:04:04 INFO Started replication on ip-10-50-2-33:3306
[martini] Completed 500 Internal Server Error in 703.560263ms

and nothing is changed on the replica.

Any ideas on how to get additional detail on this error, so we can figure out the issue?

**Comment by shlomi-noach** [comment]
Nothing jumps to mind on the error part. However, `skip-query` was not built for GTIDs. It was built in pre-GTID era. With GTIDs skipping queries is not advisable (and probably I should disable it on GTID enabled replicas).

So this is an orthogonal discussion: what kind of query were you trying to skip and why? It would probably not work and GTID will make sure the replica executes the skipped query after all.

**Comment by pmkane** [comment]
Got it.  We skipped it by hand, that's good to know re: lack of GTID support.

In this particular case, the replica was a partial replica of the master that used filtering to only handle certain databases. 
 A query was run on master while USEing a database that existed on the replica that tried to modify a table that wasn't part of the partial replica.  So that's one example of when GTID-based skips can be successful and are necessary.

**Comment by shlomi-noach** [comment]
> A query was run on master while USEing a database that existed on the replica that tried to modify a table that wasn't part of the partial replica.

In that case you want to inject an empty transaction on the replica, with the specific same GTID of that undesired event.

**Comment by pmkane** [comment]
Yup, that's how we did the skip.  I just wanted to explain why GTID skipping might still be a useful thing to support, although it's certainly far less common than in the pre-GTID days.

**Comment by shlomi-noach** [comment]
Got it. Thanks for explaining!
